### DISCUSSION TITLE
per core frequency support ?

### DISCUSSION THREAD

**Discussion by ghost** [aristocratos/btop#450] [Ideas]
Possible through `/sys/devices/system/cpu/cpu*/cpufreq/scaling_cur_freq` I guess ? It's important atleast for me, as I have to monitor per core frequency to find which core scales frequencies how.

Also, an enhancement request that's being asked a year ago ? #190.

**Comment by aristocratos** [comment]
@pc00per 
> Also, an enhancement request that's being asked a year ago ? https://github.com/aristocratos/btop/issues/190.

Are you asking why one request of 100+ open requests haven't been fulfilled?

**Reply by ghost** [reply]
Looks like I am though.
The default "overall frequency" aside cpu name is kinda useless. It always either stays at 2.9GHz or 4GHz (turbo). On catting `scaling_cur_freq`, most of the time cores remain idle (800MHz). 
I donno how you calculating the overall cpu freq, but it doesn't feel right. It doesn't look like all core average too.

**Reply by aristocratos** [reply]
> Looks like I am though.

Are you unfamiliar with how open source works?
No one is gonna spend their own unpaid time writing code you wan't unless they also have a need for it.
So you either wait for that to happen or learn to code and write it yourself.

Requests for btop are fulfilled in the rare case I have time over when there isn't bugs needing fixing and the request seems plausible and fitting, or in very rare cases when someone else writes the code and submits a pull request.

> It doesn't look like all core average too.

An average doesn't work since you have to consider offline cores, cores with different max clocks (like all new Intel cpus), etc.

The cpu clock is taken from the first available core:
https://github.com/aristocratos/btop/blob/2247ed3db10b26b38a064e05ead8f614ba88bc79/src/linux/btop_collect.cpp#L407-L462

**Reply by ghost** [reply]
Tq for the information. I do appreciate your work, & hope someone make a pr, as I'm kinda inexperienced. It's my nuisance that I couldn't able to go through the source code. I do know some programming in bash & c though, but I'm too inefficient in writing it.

Right now, I'm just manually looping `scaling_cur_freq` every second to see the exact frequencies with an alias. Then adding/averaging all those freqs to print my expected output, which happens to be 1200MHz (1.2GHz) on idle. Which is why I got confused with btop's calculations.

**Reply by aristocratos** [reply]
No problem, this request will likely be fulfilled at some point but I can't say when.

One thing to consider though is the fact that anytime btop is measuring a lot of other stuff is happening in sequence which drives up the cpu clock,
I've mitigated this in the latest commit by moving the function for getting cpu clock first in the list, which should get the clocks before they peak.

You can test it by compiling with latest commit or with binary from https://github.com/aristocratos/btop/actions/runs/3337669312 when it finishes building.

**Reply by ghost** [reply]
Great. Will try. Hoping for next releases for even better experience.
